java-generator/core/src/main/java/io/fabric8/java/generator/URLJavaGenerator.java

```diff
@@ -57,13 +57,13 @@ public void run(File outputDirectory) {
       throw new JavaGeneratorException(
           "Download directory is required");
     }
+    if (!downloadDirectory.exists() && !downloadDirectory.mkdirs()) {
+      throw new JavaGeneratorException("Unable to create download directory " + downloadDirectory.getAbsolutePath());
+    }
```

Comment:
Do you think `downloadDirectory.isDirectory()`(on line 63) check is necessary after moving this check-up? Won't `downloadDirectory.mkdirs()` returning `true` would guarantee this?

Reply:
yes, you are right, this is redundant, happy to remove the condition!

Reply:
reverted as we have a specific test looking for the folder colliding with a file, I think is fine to leave it as in my first proposal ...